sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentryRequestFilter.java

```diff
@@ -0,0 +1,36 @@
+package io.sentry.spring.boot;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.core.IHub;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/** Pushes new {@link io.sentry.core.Scope} on each incoming HTTP request. */
+@Open
+public class SentryRequestFilter extends OncePerRequestFilter {
+  private final @NotNull IHub hub;
+
+  public SentryRequestFilter(final @NotNull IHub hub) {
+    this.hub = hub;
+  }
```

Comment:
would it make sense that all those classes that require a `hub` have a package-private ctor with the `hub` param for testing purposes, but a public ctor already sets the hub from `HubAdapter.getInstance()` so nobody needs to pass an instance when creating those objects?

Reply:
It is very not-spring idiomatic to get dependencies using static methods. For anyone who worked with Spring or Java EE, dependency injection is a natural way to use dependencies, so I would say it should stay as it is.

Reply:
if all those classes are only used internally and managed by us, I'd say its fine, but if one needs to `new SentryRequestFilter(HubAdapter.getInstance())`, then it'd be a bit weird IMO.
the other way would be to refactor `HubAdapter.getInstance()` and find a better approach in a different PR, but I'm happy the way how it is

Reply:
They would not need to do `HubAdapter.getInstance()` but rather inject `IHub` that is created in auto-configuration.

Reply:
got it, we are good to go then